src/main/java/pl/onsight/wypozyczalnia/controller/AdminController.java

```diff
@@ -32,29 +38,68 @@ public ModelAndView adminPage(ModelAndView modelAndView) {
+        if (orders.contains(productService.findProductById(id))) {
+            modelAndView.addObject("info", new Info("Produkt, który chcesz usunąć znjaduję sie w zamówieniu!", true));
+        } else {
+            productService.removeProduct(id);
+            modelAndView.addObject("info", new Info("Produkt usunięto poprawnie!", true));
+            adminPage(modelAndView);
+        }
+    }
+
+
     @PostMapping("/admin-page")
     public ModelAndView addProduct(@ModelAttribute @Valid ProductEntity product,
                                    BindingResult bindingResult, ModelAndView modelAndView) {
+        List<ProductEntity> allProducts = productService.findAllProducts();
+        List<String> allNamesOfProducts = allProducts.stream().map(name -> name.getProductName()).collect(Collectors.toList());
         if (bindingResult.hasErrors()) {
             modelAndView.setViewName("admin");
+        } else if (allNamesOfProducts.contains(product.getProductName())) {
+            modelAndView.addObject("info", new Info("Produkt o nazwie " + product.getProductName() + " już istnieje!", false));
+            return adminPage(modelAndView);
         } else {
             productService.addProduct(product);
             modelAndView.setViewName("redirect:/shop");
         }
         return modelAndView;
     }
 
-    @DeleteMapping(path = "/product-order/{id}")
-    @ResponseBody
-    public void removeProductOrder(@PathVariable Long id) {
-        productOrderService.removeProductOrder(id);
+    @GetMapping("/edit/{id}")
+    public ModelAndView showEditForm(@PathVariable("id") Long id, ModelAndView modelAndView) {
+        ProductEntity product = productService.findProductById(id);
+        modelAndView.addObject("product", product);
+        modelAndView.setViewName("editProduct");
+        return modelAndView;
     }
 
-    @DeleteMapping(path = "/product/{id}")
-    @ResponseBody
-    public void removeProduct(@PathVariable Long id) {
-        productService.removeProduct(id);
+    @PostMapping("/edit/{id}")
+    public ModelAndView editProduct(@PathVariable("id") Long id, @Valid ProductEntity product,
+                                    ModelAndView modelAndView) {
+        List<ProductEntity> allProducts = productService.findAllProducts();
+        List<String> allNamesOfProducts = allProducts.stream().map(name -> name.getProductName()).collect(Collectors.toList());
+        String tempName = productService.findProductById(id).getProductName();
+        String newName = product.getProductName();
+        allNamesOfProducts.remove(tempName);
+        if (allNamesOfProducts.contains(newName)) {
+            modelAndView.addObject("info", new Info("Produkt o nazwie " + product.getProductName() + " już istnieje!", false));
+            adminPage(modelAndView);
+        } else {
+            productService.addProduct(product);
+            adminPage(modelAndView);
+        }
+        return modelAndView;
     }
+}
 
```

Comment:
wydaje mi sie ze lepiej by tu bylo zrobic metode w walidatorze ktora sprawdza, czy nazwa i id zmienianego produktu sa takie same, jesli tak to spoko (bo to znaczy ze zmieniles cos w produkcie, ale nie zmieniles nazwy, co powinno byc dopuszczalne oczywiscie), a jesli taka nazwe ma produkt z innym id to nazwa zajeta